docs/spec/spec.md

```diff
@@ -0,0 +1,416 @@
+    # The SAP client (jco.client.client)
+    string jcoClient;
+    # The SAP user name (jco.client.user)
+    string user;
+    # The SAP password (jco.client.passwd)
+    string passwd;
+    # The SAP language (jco.client.lang)
+    string lang = "EN";
+    # The SAP group (jco.client.group)
+    string group = "PUBLIC";
+};
+```
+
+**Example Configuration:**
+
+```ballerina
+configurable jco:DestinationConfig config = ?;
+jco:Client sapClient = check new (config);
+```
+
+Config.toml file:
+
+```toml
+[config]
+ashost = "host"
+sysnr = "00"
+jcoClient = "000"
+user = "user"
+passwd = "password"
+lang = "en"
+group = "group"
+```
+
+##### 2.1.1.2 AdvancedConfig
+
+If the user wants to specify more detailed configurations beyond `DestinationConfig`, they can use `AdvancedConfig`. The `AdvancedConfig` type is a map that holds any SAP JCo configurations accepted by the SAP destination provider. 
+
+```ballerina
+public type AdvancedConfig map<string>;
+```
+
+**Example Configuration:**
+
+```ballerina
+configurable jco:AdvancedConfig config = ?;
+jco:Client sapClient = check new (config);
+```
+
+Config.toml file:
+
+```toml
+[config]
+"jco.destination.pool_capacity" = "3"
+"jco.destination.peak_limit" = "10"
+```
+
+#### 2.1.2 Available Operations
+
+##### 2.1.2.1 Execute a Function Module via RFC
+
+An RFC function call is made using the following function;
+
+```ballerina
+# Executes the RFC function
+#
+# + functionName - The name of the function to be executed
+# + importParams - The input parameters for the function
+# + exportParams - The output parameters for the function
+# + return - An error if the execution fails
+remote function execute(string functionName, record {|FieldType?...;|} importParams, typedesc<record {|FieldType?...;|}|xml|json?> exportParams = <>) returns exportParams|Error
+```
+
+The input parameters require functionName and importParams. The functionName refers to the Remote Function Module name and importParams accepts a FieldType value, which can be;
+
+```ballerina
+public type FieldType string|int|float|decimal|time:Date|time:TimeOfDay|byte[]|record {|FieldType?...;|}|record {|FieldType?...;|}[];
+```
+
+For the export parameters, the type descriptor of the user's return type is extracted. It should be a closed record of `FieldType` as the rest fields or a closed record with the exact output parameter names from the SAP system. This initial record provided by the user will be considered the export parameter list, and within this closed record, the user can include nested records for `SAP structures` and record arrays for `SAP tables`.
+
+Users can invoke it as shown below:
+
+```ballerina
+public function main() returns error? {
+    ImportParams importParams = {
+        importParam1: "Hello",
+        importParam2: 123,
+        importParam3: 123.45,
+        importParam4: 123.456
+    };
+
+    ExportParams? result = check jcoClient->execute("TEST_FUNCTION", importParams);
+    if result is jco:ExportParams {
```

Comment:
Do we have an in-built record type for `ExportParams` in jco?

Reply:
No.

Reply:
We are returning the data according to the user-given `ExportParams` type.

```
remote function execute(string functionName, record {|FieldType?...;|} importParams, typedesc<record {|FieldType?...;|}|xml|json?> exportParams = <>) returns exportParams|Error
```